itest/lnd_sweep_test.go

```diff
@@ -2146,3 +2154,154 @@ func runBumpFee(ht *lntest.HarnessTest, alice *node.HarnessNode) {
+	// transactions.
+	if ht.IsNeutrinoBackend() {
+		ht.Skipf("skipping BumpForceCloseFee test for neutrino backend")
+	}
+	// fundAmt is the funding amount.
+	fundAmt := btcutil.Amount(1_000_000)
+
+	// We add a push amount because otherwise no anchor for the counter
+	// party will be created which influences the commitment fee
+	// calculation.
+	pushAmt := btcutil.Amount(50_000)
+
+	openChannelParams := lntest.OpenChannelParams{
+		Amt:     fundAmt,
+		PushAmt: pushAmt,
+	}
+
+	// Bumping the close fee rate is only possible for anchor channels.
+	cfg := []string{
+		"--protocol.anchors",
+	}
+
+	// Create a two hop network: Alice -> Bob.
+	chanPoints, nodes := createSimpleNetwork(ht, cfg, 2, openChannelParams)
+
+	// Unwrap the results.
+	chanPoint := chanPoints[0]
+	alice := nodes[0]
+
+	// We need to fund alice with 2 wallet inputs so that we can test to
+	// increase the fee rate of the anchor cpfp via two subsequent calls of
+	// the`BumpForceCloseFee` rpc cmd.
+	ht.FundCoinsP2TR(btcutil.SatoshiPerBitcoin, alice)
+
+	// Alice force closes the channel which has no HTLCs at stake.
+	_, closingTxID := ht.CloseChannelAssertPending(alice, chanPoint, true)
+	require.NotNil(ht, closingTxID)
+
+	// Alice should see one waiting close channel.
+	ht.AssertNumWaitingClose(alice, 1)
+
+	// Alice should have 2 registered sweep inputs. The anchor of the local
+	// commitment tx and the anchor of the remote commitment tx.
+	ht.AssertNumPendingSweeps(alice, 2)
+
+	// Calculate the commitment tx fee rate.
+	closingTx := ht.AssertTxInMempool(closingTxID)
+	require.NotNil(ht, closingTx)
+
+	// The default commitment fee for anchor channels is capped at 2500
+	// sat/kw but there might be some inaccuracies because of the witness
+	// signature length therefore we calculate the exact value here.
+	closingFeeRate := ht.CalculateTxFeeRate(closingTx)
+
+	// We increase the fee rate of the fee function by 100% to make sure
+	// we trigger a cpfp-transaction.
+	newFeeRate := closingFeeRate * 2
+
+	// We need to make sure that the budget can cover the fees for bumping
+	// the close transction taking also the parent fee into account.
+	// We select a large enough budget to do so.
+	// However we also want to make sure that the budget is not too large
+	// so that the delta of the fee function does not increase the feerate
+	// by a single sat hence rbfing the anchor sweep every time a new block
+	// is found.
+	//
+	// NOTE:
+	// We expect an anchor sweep with 2 inputs and 1 output because we have
+	// a single large enough input in our wallet because alice's node was
+	// funded with 1 BTC.
+	// EndFeeRate: 100_000 sats/725 wu * 1000 = 137931 sat/kw
+	// StartingFeeRate: 5000 sat/kw
+	// delta = (137931-5000)/1008 = 132 sat/kw (which is lower than
+	// 250 sat/kw) => hence we are fullfilling BIP 125 Rule 4, which is
+	// exactly what we want here to test the subsequent calling of the
+	// bumpclosefee rpc.
+	cpfpBudget := 100_000
```

Comment:
@ProofOfKeags its not super trivial to come up with the right budget value, I tried to explain it in more detail, wdyt ?

Reply:
non-blocking. If it needs to be magic, make it magic.